src/lib/libdylink.js

```diff
@@ -959,11 +1014,24 @@ var LibraryDylink = {
      * @param {number=} handle
      * @param {Object=} localScope
      */`,
-  $loadDynamicLibrary: function(libName, flags = {global: true, nodelete: true}, localScope, handle) {
+  $loadDynamicLibrary: function(libName, flags = {global: true, nodelete: true}, localScope, rpath = {parentLibPath: '', paths: []}, handle) {
 #if DYLINK_DEBUG
     dbg(`loadDynamicLibrary: ${libName} handle: ${handle}`);
     dbg(`existing: ${Object.keys(LDSO.loadedLibsByName)}`);
 #endif
+
+#if FILESYSTEM
+      var runtimePathsAbs = (rpath.paths || []).map((p) => replaceORIGIN(rpath.parentLibPath, p));
+      var searchDirs = getDefaultLibDirs().concat(runtimePathsAbs);
+      var [existsInFS, libNameAbs] = locateLibraryFromFS(libName, searchDirs);
+      if (existsInFS) {
+        libName = libNameAbs;
```

Comment:
Review point: This part will switch the `libName`  to the absolute path, which means the values stored in `LSDO.loadedLibsByName` will change to absolute path accordingly.

In Pyodide, we were already storing absolute path to prevent loading the same library twice ([code pointer](https://github.com/pyodide/pyodide/blob/a53c17fd8ef87e79c7d3bfbb1262ca7ff6718ef2/src/js/dynload.ts#L157-L168)), but I guess this can be a breaking change?

Also, the absolute path of the shared library is needed to calculate the `$ORIGIN` of the rpath.

Reply:
So the only case where it could be a problem is if they were defining `EMV['LD_LIBRARY_PATH']` and there is a file of the given name on the `LD_LIBRARY_PATH`? Otherwise there is no change? Doesn't seem like a problematic change to me.

Reply:
It would also change the parameter passed to [`locateFile`](https://github.com/emscripten-core/emscripten/blob/f391a0d8462339a22ac335a0c8e00693614748de/src/lib/libdylink.js#L1027C21-L1027C31) which will be called as a fallback when loading from filesystem fails.

But in that case, two conditions should met: 1) the file is in the filesystem, 2) but it had failed to load from the filesystem for some reason, hence fallback to `locateFile`. So I don't think it would be a breaking change for users of `locateFile`.